extensions/amp-script/0.1/amp-script.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout, isLayoutSizeDefined} from '../../../src/layout';
+import {MainThread} from '@ampproject/worker-dom/dist/index.safe.patched';
+import {
+  calculateExtensionScriptUrl,
+} from '../../../src/service/extension-location';
+import {dev} from '../../../src/log';
+import {getMode} from '../../../src/mode';
+
+/** @const {string} */
+const TAG = 'amp-script';
+
+export class AmpScript extends AMP.BaseElement {
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.CONTAINER ||
+        isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const url = this.workerThreadUrl_();
+    dev().fine(TAG, 'Fetching amp-script-worker from:', url);
+    MainThread.upgradeElement(this.element, url);
```

Comment:
This currently infers the URL of the script from the element, right? I'd prefer if we switch to an explicit API where AMP reads the attribute.